meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/TestZkMetaClient.java

```diff
@@ -67,8 +68,16 @@ public class TestZkMetaClient {
 
   private ZkServer _zkServer;
 
+  /**
+   * Creates local Zk Server
+   * Note: Cannot test container / TTL node end to end behavior as
+   * the zk server setup doesn't allow for that. However, the actual
+   * behavior has been verified to work on native ZK Client.
```

Comment:
Thanks for the change. Could you please give a bit more detail explaining why the set up dues not support container / TTL node and what are needed if we want to support that in unit test?

Reply:
+1 
Does setting this `System.setProperty("zookeeper.extendedTypesEnabled", "true");` not help?

Reply:
To enable proper end-to-end behavior between TTL and Container nodes in unit testing, it is necessary to invoke ContainerManager.java when setting up the Zk server. ContainerManager object is responsible for managing the creation / deletion of znodes (lifecycle of the nodes).

Since nodes with TTL and Container modes are designed to be deleted after a certain interval, a zk server with a ContainerManager is required for running these nodes. This can be achieved by using classes such as LeaderZooKeeperServer.java or ZooKeeperServerMain.java. However, the current implementation of zk server in metaclient and helix test base utilizes a simplified version of ZooKeeperServer, which lacks a ContainerManager implementation. 

It is necessary to create both a ContainerManager and `System.setProperty("zookeeper.extendedTypesEnabled", "true");` in order for TTL and Container nodes to function properly.

For future work, the setup of the zk server needs to be changed to use the appropriate classes mentioned above to enable the proper functioning of TTL and Container nodes in unit testing. I'll update the comments.

Reply:
Thanks for the info Marcos! 
Is this a Zookeeper side issue for the inconsistency? Do we need ZK version upgrade to have ContainerManager? Just want to understand if we have external dependency in order to test this.

Reply:
Of course! And no to both. I believe the `ZooKeeperServer `class exists in Zk just to create servers easier but with less functionalities. And we do not need to upgrade any ZK version as the classes mentioned above are available in the current version of zk that we use.

Just for additional information, I was able to successfully create a zk server using `ZooKeeperServerMain.java,` but replacing the existing framework and modifying it so it's thread safe would require more bandwidth than available. And it is not a major issue so the priority is low.